### ISSUE TITLE
Add MacOS support

### ISSUE THREAD

**Issue by slonopotamus** [adamrehn/ue4-docker#168]
There exists [Docker-OSX](https://github.com/sickcodes/Docker-OSX) project that we could use as a base for MacOS builds.

**Comment by adamrehn** [comment]
Having a read through the Docker-OSX documentation, it definitely seems like it's a useful solution for **running** macOS VM images inside Linux containers, but I'm not sure how useful it is for **building** the VM images? It doesn't look like it supports building Dockerfiles, and the examples all appear to be based around spinning up a container and then running commands to modify the state of the mutable VM image. Unless I'm misunderstanding something, it seems to me like the best approach would be to build macOS VM images with a mature image builder tool like [HashiCorp Packer](https://www.packer.io/) and then wrap them in Docker-OSX container images as a post-processing step. The end result would still be a container image, and therefore still in scope for the ue4-docker project.

@slonopotamus what are your thoughts? Are there any details that I've missed or misinterpreted?

**Comment by slonopotamus** [comment]
Well, technically building a container is just running a container and then commiting the result, so building isn't very much different from running.

I'm not sure if Docker-OSX will actually end up usable or we would instead go a completely different route. Think of it as just some input data for investigation.

**Comment by adamrehn** [comment]
> Well, technically building a container is just running a container and then commiting the result, so building isn't very much different from running.

Indeed, my thinking is mostly around how robust the tooling is around running multiple commands and capturing the result. Without Dockerfile support, Docker-OSX looks like you'd need to write your own scripting to do this, whereas tools like Packer already provide a well-tested implementation of this functionality.

> I'm not sure if Docker-OSX will actually end up usable for this task or we would instead go a completely different route. Think of it as just some input data for investigation.

Understood. It's certainly interesting to see projects like Docker-OSX providing a container interface for macOS systems in lieu of true container support.

**Comment by adamrehn** [comment]
I've had some additional thoughts about this, which I've documented here: https://github.com/containerd/containerd/discussions/5525
